pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -121,12 +139,27 @@ func NewAccessControlDashboardPermissionFilter(user *user.SignedInUser, permissi
 		}
 	}
 
-	return AccessControlDashboardPermissionFilter{user: user, folderActions: folderActions, dashboardActions: dashboardActions}
+	f := accessControlDashboardPermissionFilter{user: user, folderActions: folderActions, dashboardActions: dashboardActions, features: features,
+		recQueries:                   make([]clause, 0, maximumRecursiveQueries),
```

Comment:
super nit: I think we could skip creating the slice here and having `maximumRecursiveQueries` - just to simplify the code a bit. But no strong feelings either way.

Reply:
sorry I don't fully understand: do you mean to initialise the slice inside `buildClauses()` instead?

Reply:
Yeah, exactly - I wonder if it even needs explicit initialisation. But this is honestly super minor and probably just a matter of preference.